bindings/rust/s2n-tls/src/connection.rs

```diff
@@ -63,12 +63,6 @@ impl fmt::Debug for Connection {
 /// s2n_connection objects can be sent across threads
 unsafe impl Send for Connection {}
 
-/// # Safety
-///
-/// All C methods that mutate the s2n_connection are wrapped
-/// in Rust methods that require a mutable reference.
-unsafe impl Sync for Connection {}
```

Comment:
Do we need to remove this from the Config too? https://github.com/aws/s2n-tls/blob/main/bindings/rust/s2n-tls/src/config.rs#L27-L31

Reply:
No we want `Sync` on `Config` since it can be cloned and used in multiple places.

Reply:
I reread the Sync documentation: 
> Types for which it is safe to share references between threads.

> The precise definition is: a type T is [Sync](https://doc.rust-lang.org/stable/std/marker/trait.Sync.html) if and only if &T is [Send](https://doc.rust-lang.org/stable/std/marker/trait.Send.html). In other words, if there is no possibility of [undefined behavior](https://doc.rust-lang.org/stable/reference/behavior-considered-undefined.html) (including data races) when passing &T references between threads.

Don't we need that for the connection? Customers need to be able to call s2n_send and s2n_recv from different threads.

Reply:
Talked offline: we will need a separate `split` API to do this correctly/safely. Otherwise, you can use `tokio::io::split`, which uses a Mutex underneath.

I've opened an issue to track this: https://github.com/aws/s2n-tls/issues/4291